library/src/main/java/com/bumptech/glide/load/resource/bitmap/Downsampler.java

```diff
@@ -248,17 +248,6 @@ static void calculateScaling(DownsampleStrategy downsampleStrategy, int degreesT
     float adjustedScaleFactor = powerOfTwoSampleSize * exactScaleFactor;
 
     options.inSampleSize = powerOfTwoSampleSize;
-    // Density scaling is only supported if inBitmap is null prior to KitKat. Avoid setting
-    // densities here so we calculate the final Bitmap size correctly.
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
```

Comment:
Agreed.
I was making this PR late last night, usually not the best time :)
Afterwards I was thinking about how this could be related to the issue. My current thought is that there is a bug in the OS code of the MotoX 5.1: bitmap scaling is not thread safe.
When you specify the inDensity and inTargetDensity it will also trigger the same non-thread-safe code path.
I will validate this hunch by applying the same lock on this operation as is applied to the drawBitmap operation.